contracts/multiswap-contracts/MultiswapForge.sol

```diff
@@ -0,0 +1,147 @@
+    ) external {
+        super.withdrawSigned(
+            token,
+            payee,
+            amount,
+            salt,
+            expiry,
+            multiSignature
+        );
+    }
+
+    // Function that returns the gas estimation from backend using estimateGas()
+    function estimateGasForWithdrawSigned(
+        address token,
+        address payee,
+        uint256 amount,
+        bytes32 salt,
+        uint256 expiry,
+        bytes memory multiSignature
+    ) external {
+        // Encode the function call data with selector only
+        bytes4 selector = bytes4(
+            keccak256(
+                "withdrawSignedForGasEstimation(address,address,uint256,bytes32,uint256,bytes)"
+            )
+        );
+        bytes memory data = abi.encodeWithSelector(
+            selector,
+            token,
+            payee,
+            amount,
+            salt,
+            expiry,
+            multiSignature
+        );
+
+        (bool success, ) =   address(this).call(data);
+
+        require(success == false);
+    }
+
+    // Override and revert the 'withdrawSignedAndSwapOneInch' function
+    function withdrawSignedAndSwapOneInch(
+        address payable to,
+        uint256 amountIn,
+        uint256 amountOut,
+        address foundryToken,
+        address targetToken,
+        bytes memory oneInchData,
+        bytes32 salt,
+        uint256 expiry,
+        bytes memory multiSignature
+    ) public override {
+       revert("Not Supported");
+    }
+    
+    // This function is only used specifically for GasEstimation & Simulation of withdrawSignedAndSwapOneInch
+    function withdrawSignedAndSwapOneInchForGasEstimation(
+        address payable to,
+        uint256 amountIn,
+        uint256 amountOut,
+        address foundryToken,
+        address targetToken,
+        bytes memory oneInchData,
+        bytes32 salt,
+        uint256 expiry,
+        bytes memory multiSignature
+    ) external {
+        // Call the original function from FiberRouter
+        super.withdrawSignedAndSwapOneInch(
+            to,
+            amountIn,
+            amountOut,
+            foundryToken,
+            targetToken,
+            oneInchData,
+            salt,
+            expiry,
+            multiSignature
+        );
+    }
+
+    // Function that returns the gas estimation from backend using estimateGas()
+    function estimateGasForWithdrawSignedAndSwapOneInch(
+        address payable to,
+        uint256 amountIn,
+        uint256 amountOut,
+        address foundryToken,
+        address targetToken,
+        bytes memory oneInchData,
+        bytes32 salt,
+        uint256 expiry,
+        bytes memory multiSignature
+    ) external {
+        // Encode the function call data with selector only
+        bytes4 selector = bytes4(
+            keccak256(
+                "withdrawSignedAndSwapOneInchForGasEstimation(address,uint256,uint256,address,address,bytes,bytes32,uint256,bytes)"
+            )
```

Comment:
Can use `this.withdrawSignedAndSwapOneInchForGasEstimation.selector`

Reply:
Implemented.